data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/OnConstraintUsageRuleHelper.java

```diff
@@ -0,0 +1,97 @@
+
+    private Map<String, Set<String>> importedCnecsNativeToFaraoIdsAndCombinableWithRa = new HashMap<>();
+    private final Map<String, String> excludedAssessedElementsByRemedialAction = new HashMap<>();
+    private final Map<String, String> includedAssessedElementsByRemedialAction = new HashMap<>();
+    private final Map<String, String> consideredAssessedElementsByRemedialAction = new HashMap<>();
+
+    public OnConstraintUsageRuleHelper(Set<CsaProfileCnecCreationContext> csaProfileCnecCreationContexts, PropertyBags assessedElements, PropertyBags assessedElementsWithRemedialAction) {
+        this.csaProfileCnecCreationContexts = csaProfileCnecCreationContexts;
+        this.assessedElements = assessedElements;
+        this.assessedElementsWithRemedialAction = assessedElementsWithRemedialAction;
+        readAssessedElements();
+        readAssessedElementsWithRemedialAction();
+    }
+
+    private void readAssessedElements() {
+        importedCnecsNativeToFaraoIdsAndCombinableWithRa = getImportedCnecsNativeIdsToFaraoIds();
+
+        List<String> nativeIdsOfCnecsCombinableWithRas = assessedElements.stream()
+                .filter(element -> element.getId(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_IS_COMBINABLE_WITH_REMEDIAL_ACTION) != null && Boolean.parseBoolean(element.getId(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT_IS_COMBINABLE_WITH_REMEDIAL_ACTION)))
+                .map(element -> removePrefix(element.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT))).collect(Collectors.toList());
+
+        importedCnecsNativeToFaraoIdsAndCombinableWithRa.keySet().retainAll(nativeIdsOfCnecsCombinableWithRas);
+    }
+
+    public void readAssessedElementsWithRemedialAction() {
+        assessedElementsWithRemedialAction.stream().filter(this::checkNormalEnabled).forEach(propertyBag -> {
+            String combinationConstraintKind = propertyBag.get(CsaProfileConstants.COMBINATION_CONSTRAINT_KIND);
+            if (combinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.EXCLUDED.toString())) {
+                excludedAssessedElementsByRemedialAction.put(removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_REMEDIAL_ACTION)), removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT)));
+            } else if (combinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.INCLUDED.toString())) {
+                includedAssessedElementsByRemedialAction.put(removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_REMEDIAL_ACTION)), removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT)));
+            } else if (combinationConstraintKind.equals(CsaProfileConstants.ElementCombinationConstraintKind.CONSIDERED.toString())) {
+                consideredAssessedElementsByRemedialAction.put(removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_REMEDIAL_ACTION)), removePrefix(propertyBag.get(CsaProfileConstants.REQUEST_ASSESSED_ELEMENT)));
```

Comment:
here, you can remove all CNECs that are combinable with remedial actions. this may solve some duplications